src/SoftCascade.php

```diff
@@ -86,9 +86,14 @@ private function execute($relation)
     private function runNestedRelations($relation)
     {
         /* TO-DO: pretty sure we can do this on the query w/o get(). */
-        /* To-DO: only run withTrashed when restore is triggered. */
-        foreach ($relation->withTrashed()->get() as $model) {
-            $this->run($model);
+        if ($this->direction == 'delete') {
+            foreach ($relation->get() as $model) {
+                $this->run($model);
+            }
+        } else {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method runNestedRelations uses an else expression. Else is never necessary and you can simplify the code to work without else.](https://www.codacy.com/app/gcphost/laravel5-soft-cascade/file/2613158207/issues/source?bid=3269076&fileBranchId=3311739#l93)